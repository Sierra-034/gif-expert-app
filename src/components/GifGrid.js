import React from "react";

export const GifGrid = ({ category }) => {

    const getGifs = async () => {
        const URL = 'https://api.giphy.com/v1/gifs/search?q=Rick+and+Morty&limit=10&limit=10&api_key=4N0WEqRJUE4MR2NNALGC8qIYVvuNUXwM';
        const response = await fetch(URL);
        const {data} = await response.json();

        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
        
        console.log(gifs);
    };

    getGifs();
    
    return (
        <div>
            <h3>{ category }</h3>
        </div>
    );
};