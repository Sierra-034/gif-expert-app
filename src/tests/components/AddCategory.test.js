import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { AddCategory } from "../../components/AddCategory";

describe('Pruebas en <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);
    
    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });
    
    test('Mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Cambia la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola mundo';
        input.simulate('change', {target: {value}});

        expect(wrapper.find('p').text()).toBe(value);
    });

    test('No debe postear la informaciÃ³n onSubmit', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        expect(setCategories).not.toHaveBeenCalled();
    });
});
