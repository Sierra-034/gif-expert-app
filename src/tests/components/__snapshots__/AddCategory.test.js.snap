// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> Mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
